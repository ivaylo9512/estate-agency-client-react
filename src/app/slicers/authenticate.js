import { createSlice } from "@reduxjs/toolkit"

const initialState = {
    user: null,
    registerRequest: {
        loading: false,
        errorMessage: undefined
    },
    loginRequest: {
        loading: false,
        error: undefined
    }
}

export const authenticateSlice = createSlice({
    name: 'authenticate',
    initialState,
    reducers: {
        setUser: (state, action) => {
            state.user = action.payload;
        },
        removeUser: () => {
            state.user = null
        },
        loginRequest: (state) => {
            state.loginRequest = {
                loading: true,
                error: undefined
            }
        },
        registerRequest: (state) => {
            state.registerRequest = {
                loading: true,
                error: undefined
            }
        },
        onLoginComplete: (state, action) => {
            console.log('here')
            const { user, error } = action.payload
            state.user = user;
            state.loginRequest = {
                loading: false,
                error
            }
        },
        onRegisterComplete: (state, action) => {
            const { user, error } = action.payload
            state.user = user;
            state.registerRequest = {
                loading: false,
                error
            }
        }
    }
})

export const { setUser, removeUser, loginRequest, registerRequest, onLoginComplete, onRegisterComplete } = authenticateSlice.actions
export default authenticateSlice.reducer