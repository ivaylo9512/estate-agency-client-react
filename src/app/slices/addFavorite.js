const { createSlice } = require("@reduxjs/toolkit")

const initialState = {
    favoriteStates: new Map()
}

const addFavoriteSlice = createSlice({
    name: 'addFavorite',
    initialState,
    reducers:{
        addFavorite: (state, {payload}) => {
            state.favoriteStates.set(payload, {
                isLoading: true,
                error: null,
                isFavorite: true
            });
        },
        onAddFavoriteComplete: (state, {payload}) => {
            const favoriteState = state.favoriteStates.get(payload);
            favoriteState.isLoading = false;
        },
        onAddFavoriteError: (state, {payload: {id, error}}) => {
            const favoriteState = state.favoriteStates.get(id);
            favoriteState.isLoading = false;
            favoriteState.error = error;
            favoriteState.isFavorite = false;
        },
        resetAddFavoriteState: (state) => {
            state.favoriteStates = initialState.favoriteStates
        }
    }
})

export const {onAddFavoriteComplete, onAddFavoriteError, addFavorite, resetAddFavoriteState} = addFavoriteSlice.actions;
export default addFavoriteSlice.reducer;
export const getAddFavoriteState = state => state.addFavorite;