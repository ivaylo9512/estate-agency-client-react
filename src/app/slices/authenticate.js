import { createSlice } from "@reduxjs/toolkit"

const user = typeof window != 'undefined' && localStorage.getItem('user') 
    ? JSON.parse(localStorage.getItem('user'))
    : undefined;

const initialState = {
    user,
    isAuth: !!user,
    registerRequest: {
        isLoading: false,
        error: null
    },
    loginRequest: {
        isLoading: false,
        error: null
    }
}

const authenticateSlice = createSlice({
    name: 'authenticate',
    initialState,
    reducers: {
        loginRequest: (state) => {
            state.loginRequest.isLoading = true;
            state.loginRequest.error = null;
        },
        registerRequest: (state) => {
            state.registerRequest.isLoading = true;
            state.registerRequest.error = null;
        },
        onLoginComplete: (state, {payload}) => {
            state.user = payload
            state.isAuth = true;
            state.loginRequest.isLoading = false;
            state.loginRequest.error = null;
        },
        onLoginError: (state, {payload}) => {
            state.loginRequest.isLoading = false;
            state.loginRequest.error = payload;
        },
        onRegisterComplete: (state, {payload}) => {
            state.user = payload
            state.isAuth = true;
            state.registerRequest.isLoading = false;
            state.registerRequest.error = null;
        },
        onRegisterError: (state, {payload}) => {
            state.registerRequest.isLoading = false;
            state.registerRequest.error = payload;
        },
        onLogout: (state, {payload}) => {
            state.loginRequest.error = payload
            state.user = undefined;
            state.isAuth = false;
        },
    }
})

export const { removeUser, loginRequest, registerRequest, onLoginComplete, onLoginError, onRegisterComplete, onRegisterError } = authenticateSlice.actions
export default authenticateSlice.reducer

export const getLoginRequest = state => state.authenticate.loginRequest;
export const getRegisterRequest = state => state.authenticate.registerRequest;
export const getUser = state => state.authenticate.user;
export const getIsAuth = state => state.authenticate.isAuth;