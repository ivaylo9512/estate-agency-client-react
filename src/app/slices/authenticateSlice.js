import { createSlice } from "@reduxjs/toolkit"

export const getLoggedUser = () => typeof window != 'undefined' && localStorage.getItem('user') 
    ? JSON.parse(localStorage.getItem('user')) 
    : null;
    
const user = getLoggedUser();

const initialState = {
    user,
    isAuth: !!user,
    registerRequest: {
        isLoading: false,
        error: null
    },
    loginRequest: {
        isLoading: false,
        error: null
    }
}

const authenticateSlice = createSlice({
    name: 'authenticate',
    initialState,
    reducers: {
        loginRequest: (state) => {
            state.loginRequest.isLoading = true;
            state.loginRequest.error = null;
        },
        registerRequest: (state) => {
            state.registerRequest.isLoading = true;
            state.registerRequest.error = null;
        },
        onLoginComplete: (state, {payload}) => {
            state.user = payload
            state.isAuth = true;
            state.loginRequest.isLoading = false;
            state.loginRequest.error = null;
        },
        onLoginError: (state, {payload}) => {
            state.loginRequest.isLoading = false;
            state.loginRequest.error = payload;
        },
        onRegisterComplete: (state, {payload}) => {
            state.user = payload
            state.isAuth = true;
            state.registerRequest.isLoading = false;
            state.registerRequest.error = null;
        },
        onRegisterError: (state, {payload}) => {
            state.registerRequest.isLoading = false;
            state.registerRequest.error = payload;
        },
        onLogout: (state, {payload}) => {
            state.registerRequest = initialState.registerRequest
            state.loginRequest = {
                ...initialState.loginRequest,
                error: payload
            }
            state.user = null;
            state.isAuth = false;
        },
    }
})

export const { removeUser, loginRequest, registerRequest, onLoginComplete, onLoginError, onRegisterComplete, onRegisterError, onLogout } = authenticateSlice.actions
export default authenticateSlice.reducer

export const getLoginRequest = state => state.authenticate.loginRequest;
export const getRegisterRequest = state => state.authenticate.registerRequest;
export const getUser = state => state.authenticate.user;
export const getIsAuth = state => state.authenticate.isAuth;