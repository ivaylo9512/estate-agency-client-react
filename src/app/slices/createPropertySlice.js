import { createSlice } from '@reduxjs/toolkit'

const initialState = {
    isLoading: false,
    error: null,
    property: null
}

const createPropertySlice = createSlice({
    name: 'createProperty',
    initialState,
    reducers: {
        createPropertyRequest: (state) => {
            state.isLoading = true;
            state.error = null;
            state.property = null;
        },
        onCreatePropertyComplete: (state, { payload }) => {
            state.isLoading = false;
            state.property = payload;
            state.error = null;
        },
        onCreatePropertyError: (state, { payload }) => {
            state.isLoading = false;
            state.error = payload;
        },
        resetCreatePropertyState: (state) => {
            state.isLoading = initialState.isLoading;
            state.error = initialState.error;
            state.property = initialState.property;
        }
    }
})

export const { createPropertyRequest, onCreatePropertyComplete, onCreatePropertyError, resetCreatePropertyState } = createPropertySlice.actions
export default createPropertySlice.reducer;

export const getCreatePropertyState = state => state.createProperty