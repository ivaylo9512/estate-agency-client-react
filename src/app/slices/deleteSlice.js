import { createSlice } from '@reduxjs/toolkit';

const initialState = {
    data: {}
}

const deleteSlice = createSlice({
    name: 'deleteProperty',
    initialState,
    reducers: {
        deleteProperty: (state, { payload: id }) => {
            let request = state.data[id] ? state.data[id] : state.data[id] = {}; 
            
            request.isLoading = true; 
            request.error = null;
        },
        onDeleteComplete: (state, { payload: id }) => {
            const request = state.data[id];

            request.isLoading = false;
            request.isDeleted = true;
        },
        onDeleteError: (state, { payload: { id, error }}) => {
            const request = state.data[id];

            request.isLoading = false;
            request.error = error;
        },
        resetDelete: (state) => {
            state.data = initialState.data;
        }
    }
})

export const { deleteProperty, onDeleteComplete, onDeleteError, resetDelete } = deleteSlice.actions;
export default deleteSlice.reducer;