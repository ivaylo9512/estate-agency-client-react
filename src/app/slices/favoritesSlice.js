import { createSlice } from '@reduxjs/toolkit'

const initialState = {
    favorites: [],
    isLoading: false,
    error: null
}

const favoritesSlice = createSlice({
    name: 'favorites',
    initialState,
    reducers: {
        getFavorites: (state) => {
            state.isLoading = true;
            state.error = null;
        },
        onFavoritesComplete: (state, {payload}) => {
            state.favorites = payload;
            state.isLoading = false;
            state.error = null;
        },
        onFavoritesError: (state, { payload }) => {
            state.isLoading = false;
            state.error = payload
        },
        resetFavorites: (state) => {
            state.error = initialState.error;
            state.isLoading = initialState.isLoading;
            state.favorites = initialState.favorites
        },
    }
})

export const { getFavorites, onFavoritesComplete, onFavoritesError, resetFavorites } = favoritesSlice.actions;
export default favoritesSlice.reducer;

export const getFavoritesState = state => state.favorites;