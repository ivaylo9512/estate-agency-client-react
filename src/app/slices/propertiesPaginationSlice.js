import { createSlice } from '@reduxjs/toolkit'

const initialState = {
    dataInfo: {
        pages: 0,
        maxPages: 0,
        data: [],
        lastData: null,
        currentData: null,
        currentPage: 1,
    },
    query: {
        take: 2,
        direction: 'ASC',
        bedrooms: 0,
        location: '',
        maxPrice: 0
    },
    isLoading: false,
    isInitial: true,
    error: null,
}

const propertiesPaginationSilce = createSlice({
    name: 'propertiesPagination',
    initialState,
    reducers: {
        getProperties(state){
            state.error = null
            state.isLoading = true;
        },
        onPropertiesComplete(state, {payload: { pageable, query }}){
            state.query = query;
            state.dataInfo.maxPages = state.dataInfo.pages + pageable.pages;
            state.dataInfo.pages = state.dataInfo.pages + pageable.data.length;
            state.dataInfo.currentPage = state.dataInfo.pages;
            state.dataInfo.lastData = pageable.lastProperty;
            state.dataInfo.data = [...state.dataInfo.data, ...pageable.data];
            state.dataInfo.currentData = pageable.data[pageable.data.length - 1] || [];
            state.isLoading = false;
            state.isInitial = false;
            state.error = null
        },
        onPropertiesError(state, { payload }){
            state.isLoading = false
            state.error = payload
        },
        setPropertiesDirection(state, { payload }){
            state.query = {
                ...initialState.query,
                direction: payload
            }
            state.dataInfo = initialState.dataInfo
            state.error = initialState.error;
            state.isInitial = initialState.isInitial;
            state.isLoading = initialState.isLoading;
        },
        setCurrentProperties(state, { payload: { currentData, currentPage } }){
            state.dataInfo.currentData = currentData;
            state.dataInfo.currentPage = currentPage;
        },
        resetPropertiesState(state){
            state.isLoading = initialState.isLoading;
            state.isInitial = initialState.isInitial;
            state.dataInfo = initialState.dataInfo
            state.query = initialState.query;
            state.error = initialState.error;
        },
    }
})
export const { resetPropertiesState, onPropertiesComplete, onPropertiesError, getProperties, setPropertiesDirection, setCurrentProperties } = propertiesPaginationSilce.actions
export default propertiesPaginationSilce.reducer

export const getPropertiesData = state => state.propertiesPagination.dataInfo;
export const getPropertiesQuery = state => state.propertiesPagination.query;
export const getPropertiesState = state => state.propertiesPagination;