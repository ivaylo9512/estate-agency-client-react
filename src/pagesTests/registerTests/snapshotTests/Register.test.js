import React from 'react';
import Register from '../../../pages/register';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { getRegisterRequest } from '../../../app/slices/authenticate';

jest.mock('react-redux', () => ({
    useSelector: jest.fn(fn => fn()),
    useDispatch: () => jest.fn(),
}));

jest.mock('../../../app/slices/authenticate');

const createWrapper = (value) => {
    getRegisterRequest.mockReturnValue(value);
    return shallow(<Register />)
}
describe('RegisterSnapshotTests', () => {
    it('renders correctly page 0', () => {
        const wrapper = createWrapper({ isLoading: false, error: null })
        expect(toJson(wrapper)).toMatchSnapshot();
    });
})