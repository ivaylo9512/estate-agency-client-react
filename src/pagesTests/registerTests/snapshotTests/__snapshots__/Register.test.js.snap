// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterSnapshotTests renders correctly page 0 1`] = `
<section>
  <form
    onSubmit={[Function]}
  >
    <InputWithError
      input={
        <input
          autoComplete="username"
          maxLength={20}
          minLength={8}
          name="username"
          onChange={[Function]}
          placeholder="username"
          required={true}
          type=""
          value=""
        />
      }
    />
    <InputWithError
      input={
        <usePasswordInput__Container>
          <input
            autoComplete="new-password"
            maxLength={22}
            minLength={10}
            name="password"
            onChange={[Function]}
            placeholder="password"
            required={true}
            type="password"
            value=""
          />
          <usePasswordInput__Button
            onClick={[Function]}
            type="button"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    576,
                    512,
                    Array [],
                    "f06e",
                    "M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z",
                  ],
                  "iconName": "eye",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
          </usePasswordInput__Button>
        </usePasswordInput__Container>
      }
    />
    <InputWithError
      input={
        <usePasswordInput__Container>
          <input
            autoComplete="new-password"
            name="repeat-password"
            onChange={[Function]}
            placeholder="repeat"
            required={true}
            type="password"
            value=""
          />
          <usePasswordInput__Button
            onClick={[Function]}
            type="button"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    576,
                    512,
                    Array [],
                    "f06e",
                    "M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z",
                  ],
                  "iconName": "eye",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
          </usePasswordInput__Button>
        </usePasswordInput__Container>
      }
    />
    <InputWithError
      input={
        <input
          autoComplete="email"
          name="email"
          onChange={[Function]}
          placeholder="email"
          required={true}
          type="email"
          value=""
        />
      }
    />
    <button
      type="submit"
    >
      next
    </button>
    <span>
      Already have an account?
      <Link
        href="/login"
      >
         Log in.
      </Link>
    </span>
  </form>
</section>
`;

exports[`RegisterSnapshotTests renders correctly page 1 1`] = `
<section>
  <form
    onSubmit={[Function]}
  >
    <InputWithError
      input={
        <input
          autoComplete="username"
          maxLength={20}
          minLength={8}
          name="username"
          onChange={[Function]}
          placeholder="username"
          required={true}
          type=""
          value=""
        />
      }
    />
    <InputWithError
      input={
        <usePasswordInput__Container>
          <input
            autoComplete="new-password"
            maxLength={22}
            minLength={10}
            name="password"
            onChange={[Function]}
            placeholder="password"
            required={true}
            type="password"
            value=""
          />
          <usePasswordInput__Button
            onClick={[Function]}
            type="button"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    576,
                    512,
                    Array [],
                    "f06e",
                    "M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z",
                  ],
                  "iconName": "eye",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
          </usePasswordInput__Button>
        </usePasswordInput__Container>
      }
    />
    <InputWithError
      input={
        <usePasswordInput__Container>
          <input
            autoComplete="new-password"
            name="repeat-password"
            onChange={[Function]}
            placeholder="repeat"
            required={true}
            type="password"
            value=""
          />
          <usePasswordInput__Button
            onClick={[Function]}
            type="button"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    576,
                    512,
                    Array [],
                    "f06e",
                    "M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z",
                  ],
                  "iconName": "eye",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
          </usePasswordInput__Button>
        </usePasswordInput__Container>
      }
    />
    <InputWithError
      input={
        <input
          autoComplete="email"
          name="email"
          onChange={[Function]}
          placeholder="email"
          required={true}
          type="email"
          value=""
        />
      }
    />
    <button
      type="submit"
    >
      next
    </button>
    <span>
      Already have an account?
      <Link
        href="/login"
      >
         Log in.
      </Link>
    </span>
  </form>
</section>
`;

exports[`RegisterSnapshotTests renders correctly with passed error props with page 0 1`] = `
<section>
  <form
    onSubmit={[Function]}
  >
    <InputWithError
      error="Username is already taken."
      input={
        <input
          autoComplete="username"
          maxLength={20}
          minLength={8}
          name="username"
          onChange={[Function]}
          placeholder="username"
          required={true}
          type=""
          value=""
        />
      }
    />
    <InputWithError
      error="Password must be between 10 and 20 characters."
      input={
        <usePasswordInput__Container>
          <input
            autoComplete="new-password"
            maxLength={22}
            minLength={10}
            name="password"
            onChange={[Function]}
            placeholder="password"
            required={true}
            type="password"
            value=""
          />
          <usePasswordInput__Button
            onClick={[Function]}
            type="button"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    576,
                    512,
                    Array [],
                    "f06e",
                    "M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z",
                  ],
                  "iconName": "eye",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
          </usePasswordInput__Button>
        </usePasswordInput__Container>
      }
    />
    <InputWithError
      error="Password must be between 10 and 20 characters."
      input={
        <usePasswordInput__Container>
          <input
            autoComplete="new-password"
            name="repeat-password"
            onChange={[Function]}
            placeholder="repeat"
            required={true}
            type="password"
            value=""
          />
          <usePasswordInput__Button
            onClick={[Function]}
            type="button"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    576,
                    512,
                    Array [],
                    "f06e",
                    "M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z",
                  ],
                  "iconName": "eye",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
          </usePasswordInput__Button>
        </usePasswordInput__Container>
      }
    />
    <InputWithError
      error="Email is already taken."
      input={
        <input
          autoComplete="email"
          name="email"
          onChange={[Function]}
          placeholder="email"
          required={true}
          type="email"
          value=""
        />
      }
    />
    <button
      type="submit"
    >
      next
    </button>
    <span>
      Already have an account?
      <Link
        href="/login"
      >
         Log in.
      </Link>
    </span>
  </form>
</section>
`;

exports[`RegisterSnapshotTests renders correctly with passed error props with page 1 1`] = `
<section>
  <form
    onSubmit={[Function]}
  >
    <InputWithError
      input={
        <input
          name="name"
          onChange={[Function]}
          placeholder="name"
          required={true}
          type=""
          value=""
        />
      }
    />
    <InputWithError
      error="location is required."
      input={
        <input
          name="location"
          onChange={[Function]}
          placeholder="location"
          required={true}
          type=""
          value=""
        />
      }
    />
    <InputWithError
      error="Description is required."
      input={
        <input
          name="description"
          onChange={[Function]}
          placeholder="description"
          required={true}
          type=""
          value=""
        />
      }
    />
    <button
      onClick={[Function]}
    >
      back
    </button>
    <button>
      register
    </button>
  </form>
</section>
`;
